### ISSUE TITLE
Frontend: page to handle async payment validations

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#4501]
For some payment methods like PayPal (when using a bank account) or the Giving Block (crypto), the payment sometimes is an async process that can take a few minutes to confirm and is only validated by a webhook event.

Today, we handle this a bit naively: we create the order as `NEW` and show the success page immediately. The order is then updated when we receive the proper webhook event. This approach is ok because most PayPal orders are in fact processed immediately.

Explained with a schema, a "regular" payment (Stripe) looks like this:
![image](https://user-images.githubusercontent.com/1556356/127136851-0aa36051-d0d6-488d-a0e6-e0e63ce21b55.png)

On the other hand, this is how an async payment should look like:
![image](https://user-images.githubusercontent.com/1556356/127138440-c52c65f1-1e4e-4040-9c6d-989bec60d244.png)

The `Processing payment...` block is the page we want to create. This page should:
- Show a loading state
- Look for an update from the API, either by repeatedly asking the status to the API or by using a WebSocket
- Redirect to the success page if the payment gets processed successfully
- Show the error if something wrong happens
- Handle timeouts

**Comment by Betree** [comment]
Ping @SudharakaP, I don't know if that's something we want for the initial release of Crypto, but let's keep this issue in mind to build a solution that will work for all systems where the payment is asynchronous.

**Comment by SudharakaP** [comment]
> Ping @SudharakaP, I don't know if that's something we want for the initial release of Crypto, but let's keep this issue in mind to build a solution that will work for all systems where the payment is asynchronous.

Thanks, I'll keep this in mind when working on Crypto. 🤔

**Comment by Betree** [comment]
cc @hdiniz I think the latest Stripe integration has some async too? Would that issue be useful?

**Comment by hdiniz** [comment]
> cc @hdiniz I think the latest Stripe integration has some async too? Would that issue be useful?

Yes, payments from the stripe payment element are mostly async with the exception of cards.
